src/feed/Feed.js

```diff
@@ -395,6 +398,7 @@ class Feed {
 
         if (
             this.endEpoch &&
+            this._stx.chart.dataSet.slice(-1)[0] &&
```

Comment:
should we check if `this._stx.chart.dataSet` is defined?

Reply:
I can add this check too, however we should also check if **this._stx.chart.dataSet.slice(-1)[0]** is defined because at the time when we get the console error, **this._stx.chart.dataSet.slice(-1)** is defined. Do you want me to add the check for this._stx.chart.dataSet as well despite this fact? :)

Reply:
Okay, it's fine if `dataSet` is not undefined at any point.